### DISCUSSION TITLE
After full disk "access memory outside buffer bounds" error

### DISCUSSION THREAD

**Discussion by n1smithy** [Ylianst/MeshCentral#6591] [Q&A] [answered]
Hello all,

I've got a problem with my docker-compose-based meshcentral installation (with mongodb):

this is the error in mesherror.txt:

```
-------- 12/6/2024, 11:52:10 PM ---- 1.1.35 --------

node:internal/buffer:86
    throw new ERR_BUFFER_OUT_OF_BOUNDS();
    ^

RangeError [ERR_BUFFER_OUT_OF_BOUNDS]: Attempt to access memory outside buffer bounds
    at boundsError (node:internal/buffer:86:11)
    at Buffer.readUInt32LE (node:internal/buffer:222:5)
    at readResourceTable (/opt/meshcentral/meshcentral/authenticode.js:555:33)
    at openFile (/opt/meshcentral/meshcentral/authenticode.js:323:29)
    at Object.createAuthenticodeHandler (/opt/meshcentral/meshcentral/authenticode.js:2059:13)
    at CreateMeshCentralServer.obj.signMeshAgents (/opt/meshcentral/meshcentral/meshcentral.js:3279:71)
    at CreateMeshCentralServer.obj.StartEx4 (/opt/meshcentral/meshcentral/meshcentral.js:1864:13)
    at CreateMeshCentralServer.obj.StartEx3 (/opt/meshcentral/meshcentral/meshcentral.js:1812:17)
    at /opt/meshcentral/meshcentral/meshcentral.js:1724:21
    at obj.GetMeshServerCertificate (/opt/meshcentral/meshcentral/certoperations.js:1091:38) {
  code: 'ERR_BUFFER_OUT_OF_BOUNDS'
}

Node.js v20.15.1
```

Anybody an idea how to repair?

**Comment by si458** [comment]
This looks like it had problems accessing either the meshagents to codesign them or accessing ur ssl certs

But it could also be you ran out of memory too?

What docker image are you using?

**Reply by n1smithy** [reply]
Just updated yesterday the official docker-compose-image.

**Reply by si458** [reply]
But u don't say what the image/tag you are using? 
The master tag is updated everytime we do a commit where as latest is every new release

**Reply by n1smithy** [reply]
Sorry, I am a little bit sleepless. Do you mean this?
ghcr.io/ylianst/meshcentral:master
with mongo:latest

**Reply by si458** [reply]
Yes, using `master` is not advised as this gets updated every time we push a commit, it's meant to be used for testing,
You should stick to using `latest` instead!

**Reply by n1smithy** [reply]
Oh, by the way: there should be enough RAM left (5 GiB free). The problem occured after the filesystem ran out of space and the containers couldn't be restartet (mongodb couldn't).

**Reply by si458** [reply]
@n1smithy so stupid one, have u increased ur disk space if u ran out?

**Reply by n1smithy** [reply]
Just had to prune docker.

**Reply by n1smithy** [reply]
Hm, I really don't know what could be wrong with those certs. They worked fine. Aw man!

EDIT: Oh, by the way: Switching to latest didn't make it. :-(

**Reply by n1smithy** [reply, marked as answer]
SOLVED it!

Just deleted the certificates from my (persistent) data-directory and restartet the containers. New certificates were built and the webcert pulled. Phew!

**Comment by n1smithy** [comment]
Anybody an idea how to restore meshcentral on docker-compose from a backup?
